packages/grafana-data/src/types/query.ts

```diff
@@ -52,3 +52,62 @@ export interface DataQuery {
    */
   datasource?: DataSourceRef | null;
 }
+
+/**
+ * Abstract representation of any label-based query
+ * @internal
+ */
+export interface AbstractQuery extends DataQuery {
```

Comment:
Would it make sense to name this `AbstractLabelQuery` instead? Since it sounds like what is is based on the comment above.

Reply:
Or maybe just `LabelQuery`

Reply:
First I named it `LabelBasedQuery` but I thought it's too long and maybe it should also include metric name, though the metric name is not really used at the moment. `LabelQuery` kind of suggests it could also be a query to "query **for** labels" (?). But I agree `AbstractQuery` seems to generic... How about `LabeledQuery`? Or going back to `LabelBasedQuery`? Or... `QueryWithLabels`? It's also loosely based on / inspired by OpenMetrics standard (though OpenMetrics standardizes metrics representation, not query abstraction itself), so maybe `OpenMetricsQuery`? 

@grafana/observability-squad what do you think?

Reply:
Hmmm, this is a tough one.  I don't think `LabelBasedQuery` is too long, but on the other side, if we want to expand it in the future to also include metrics/something else, this might be too inclusive. I kind of like that `AbstractQuery` is generic, but maybe it is too generic. I like the idea of `OpenMetricsQuery` but I am wondering if it might not be confusing with actual OpenMetrics. So I am thinking if we can come up with something unique for grafana. From all of the options, I like the `AbstractQuery` the most.

Reply:
do you think this also should include a generic type for the source query (logs, traces, metrics, etc)? some data sources support multiple types of queries, and i'm wondering if it would be useful to have the additional context about the source query when transforming to the target query language

Reply:
Okay, it's internal anyway for now so I'll keep it `AbstractQuery` but we can change it in the future.

@toddtreece Yes, I think this could include generic source query type, though we don't have a use case for it now. I think it'd be useful in cases when switching between two data sources that support multiple data types. I was also thinking that the abstract query could contain the entire original/raw query (or info about the syntax/language) to allow data sources to cover some edge cases. For example InfluxQL and SQL have very similar syntax so some elements of the query could be ported between them. The interface is marked as internal for now, I think it may change when we add more data source to it.